lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -137,13 +139,16 @@ def generate_secret_yaml(secret_name, secret_type, data)
         encoded[secret_key] = Base64.strict_encode64(value)
       end
 
+      labels = { "name" => secret_name }
+      labels.merge!(@selector) if @selector
```

Comment:
Hmm that makes sense. We'll have to support ERB for EJSON then too, because you'd want to be able to prefix secret names dynamically to ensure they're not overwritten by other deployments.

Reply:
I just did some testing and I'm afraid this isn't really going to work. If we have it look something like this:


```json
// secrets.ejson.erb
{
  "_public_key": "65f79806388144edf800bf9fa683c98d3bc9484768448a275a35d398729c892a",
  "kubernetes_secrets": {
    "<%= branch %>-catphotoscom": {
      "_type": "kubernetes.io/tls",
      "_labels": {
        "branch": "<%= branch %>"
      },
      "data": {
        "tls.crt": "EJ[1:N0RQ6a1BcGJ7w/1ABQUC3C2DQMP+qWuc10LoNUQOiUg=:Z3AW84E4MTa3ntO9jHqVAFy3cpDVF3tC:BpMYh76OGkBIi0BKlFG1gZnkSGGCmwLUX5Fk+BkUCn0mN/X5Nrvs]",
        "tls.key": "EJ[1:N0RQ6a1BcGJ7w/1ABQUC3C2DQMP+qWuc10LoNUQOiUg=:W9F16TEKrtF1rdXaBaRv6bX7uIOB0SJi:O6kgaj5mly05JXjFt10eo6sj6ZBNzvNxQrPBBQ1i0g==]"}
    }
  }
}
```

Then the problem is that ejson will encrypt `"branch": "<%= branch %>"`, even though it's under an underscore key. Probably a bug worth fixing in ejson.

Reply:
I think that's because underscore keys don't propagate downwards; you'd need to use `_branch` rather than `_labels`. Thinking about this further, I'm not 100% sure I like the idea of being able to use ERB into a secrets file. But we do need to have some way of making the behaviour predictable, including if two selector-based deploys were to happen at the same time to the same namespace. Perhaps an MVP option would be to automatically disable pruning if a selector is given, and print a warning, e.g. "Secret pruning is disabled when deploying with `--selector` because labels are not yet supported by the secret management feature." Wdyt?

Reply:
As a work-around could we include the selector `json` as a string literal to be parsed? For example, instead of:

```
{
  "_public_key": "65f79806388144edf800bf9fa683c98d3bc9484768448a275a35d398729c892a",
  "kubernetes_secrets": {
    "foo": {
      "type": "kubernetes.io",
      "_labels": {
        "branch": "foo"
      }
    }
  }
}
```

We can use:

```
{
  "_public_key": "65f79806388144edf800bf9fa683c98d3bc9484768448a275a35d398729c892a",
  "kubernetes_secrets": {
    "foo": {
        "type": "EJ[1:OrPRTUi2rTBE6Lfci5FHIoJSYK36dBwC7JYU7g8QzWw=:00UAkBOnyrR57I5Zc2oT0f8QStFHSCHb:6hQz9FzuUV0z18gjxv2028Csw8kQOFDDZM6whuI=]",
        "_labels": "{\"branch\": \"foo\"}"
    }
  }
}
```

Reply:
> ...  you'd need to use _branch rather than _labels

Just tried this out and it works, seems to be the better idea. 

I'm not opposed to ERB secrets, I don't see why template variables should be allowed in other resources, but not secrets.